wandb/filesync/dir_watcher.py

```diff
@@ -46,7 +47,7 @@ def __init__(
     ) -> None:
         self.file_path = file_path
         # Convert windows paths to unix paths
-        save_name = SaveName(util.to_forward_slash_path(save_name))
```

Comment:
Nonblocking suggestion: I think SaveName is conceptually a subtype of LogicalPath -- you could change it to `SaveName = NewType("SaveName", LogicalPath)` if you wanted.

Reply:
This was saving it for the next PR; though that re-uses it rather than creates a separate alias. Are they different enough to justify having a subtype?